packages/riverpod_analyzer_utils_tests/test/ref_invocation_test.dart

```diff
@@ -411,6 +411,49 @@ final provider = Provider<int>((ref) {
     );
   });
 
+  testSource('Decodes nested ref.read invocations with family providers',
+      runGenerator: true, source: '''
+import 'package:riverpod/riverpod.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+part 'foo.g.dart';
+
+final dep = FutureProvider((ref) => 0);
+final dep2 = FutureProvider.family((ref, int arg) => 0);
+
+final provider = Provider<int>((ref) {
+  ref.read(dep2(ref.read(dep)));
```

Comment:
Do you mind adding a tsst with a 3rd level of nesting?

Just to make sure we have a proper recursion rather than handling a single level of search.
Otherwise LGMT!